atlasdb-dbkvs/src/test/java/com/palantir/atlasdb/keyvalue/dbkvs/impl/postgres/PostgresVersionCheckTest.java

```diff
@@ -32,6 +32,8 @@
     @SuppressWarnings(value = "Slf4jConstantLogMessage")
     public void shouldLogErrorOn_9_2_24() {
         Logger log = Mockito.mock(Logger.class);
+        thrown.expect(AssertionError.class);
+        thrown.expectMessage("The minimum supported version is ");
```

Comment:
this test doesn't seem right anymore. Lines 37-40 are now dead code, right?

Reply:
assertAndLog is named oddly; the logging happens first and regardless of the assert failure. So checking the logging still passes, and it's useful for to check because assertions won't usually be enabled in an actual production environment.

Reply:
I think James's point was that if an assertion error is thrown, we wouldn't run the rest of the body of the method?

I don't think OSS Atlas runs its tests with assertions enabled, so testing this might be a little less straightforward.